tensorflow/python/kernel_tests/math_ops/segment_reduction_ops_test.py

```diff
@@ -119,7 +124,8 @@ def testValues(self, dtype):
                 indices, np_x, np_op1, np_op2, initial_value=initial_value)
             s = tf_op(data=tf_x, segment_ids=indices)
             tf_ans = self.evaluate(s)
-            self.assertAllClose(np_ans, tf_ans)
+            tol = 1e-6 if dtype != dtypes_lib.bfloat16 else 1e-2
+            self.assertAllClose(np_ans, tf_ans, rtol=tol, atol=tol)
```

Comment:
There is a `assertAllCloseAccordingToType(...)` that may work here instead of explicitly setting tolerances.